src/Symfony/Component/Security/Guard/Authenticator/AbstractFormLoginAuthenticator.php

```diff
@@ -41,9 +41,14 @@
      * login page directly), this returns the URL the user should be redirected
      * to after logging in successfully (e.g. your homepage).
      *
+     * @deprecated Implement onAuthenticationSuccess() instead of needing this function.
+     *
      * @return string
      */
-    abstract protected function getDefaultSuccessRedirectUrl();
+    protected function getDefaultSuccessRedirectUrl()
```

Comment:
I agree with @fabpot. I guess we should remove the method here. Then we can check below if the method exists, trigger a deprecation in that case and optionally call it (not sure right now if that is needed for backwards compatibility).